src/realm/object-store/sync/sync_session.cpp

```diff
@@ -1236,12 +1257,79 @@ void SyncSession::update_configuration(SyncConfig new_config)
         REALM_ASSERT(m_state == State::Inactive || m_state == State::Paused);
         REALM_ASSERT(!m_session);
         REALM_ASSERT(m_config.sync_config->user == new_config.user);
+        // Since this is used for testing purposes only, just update the current sync_config
         m_config.sync_config = std::make_shared<SyncConfig>(std::move(new_config));
         break;
     }
     revive_if_needed();
 }
 
+void SyncSession::update_sync_config_after_migration()
+{
+    // Migration state changed - Update the configuration to
+    // match the new sync mode.
+    bool flx_sync_requested;
+    {
+        util::CheckedLockGuard cfg_lock(m_config_mutex);
+        m_config.sync_config = m_migration_store->convert_sync_config(m_original_sync_config);
+        flx_sync_requested = m_config.sync_config->flx_sync_requested;
+    }
+
+    util::CheckedLockGuard cfg_lock(m_state_mutex);
```

Comment:
can we call this state_lock or something since it's not locking the config mutex?